platform/linux-generic/odp_ipsec_sad.c

```diff
@@ -956,7 +974,8 @@ int _odp_ipsec_sa_replay_precheck(ipsec_sa_t *ipsec_sa, uint32_t seq,
 				  odp_ipsec_op_status_t *status)
 {
 	/* Try to be as quick as possible, we will discard packets later */
-	if (ipsec_sa->antireplay && ((seq + ipsec_sa->in.ar.win_size) <=
+	if ((!ipsec_sa->esn) && ipsec_sa->antireplay &&
+	    ((seq + ipsec_sa->in.ar.win_size) <=
```

Comment:
This change breaks the anti-replay checking logic. After the change, replayed packets whose sequence number is below the window may get accepted.

Reply:
>After the change, replayed packets whose sequence number is below the window may get accepted.
Yes, if ESN is enabled. In this case these packets will be part of the next 32-bit AR window( not the current window). As per the ESN, these packets are supposed to be allowed.

Reply:
Sorry, I pasted my comment to a wrong piece of code. My comments was for similar change in ipsec_wslarge_replay_update(). There the change lets through old packets that fall below the window, breaking anti-replay protection.

Reply:
Ack, fixed it.